simplesat/pool.py

```diff
@@ -47,7 +49,10 @@ def add_repository(self, repository):
             self._id += 1
             self._id_to_package_[current_id] = package
             self._package_to_id_[package] = current_id
-            self._packages_by_name_[package.name].append(package)
+            for constraints in package.provides:
+                req = Requirement.from_constraints(constraints)
+                assert not req.has_any_version_constraint
```

Comment:
This does sound like an `InvalidConstraint`. If this exception trickles up to users, we would want to make sure that the message was clear, too